examples/compiled/bar_grouped.vg.json

```diff
@@ -144,7 +144,14 @@
                 },
                 "offset": 10,
                 "orient": "top",
-                "style": "guide-label"
+                "style": "guide-label",
+                "encode": {
```

Comment:
We can actually use `title` directive's `baseline` property.
However, it seems like this is not the problem introduced by this PR. Thus, let's refactor this in a separate PR.  (Perhaps as a part of https://github.com/vega/vega-lite/pull/3888?)